ui/client/components/toolbars/edit/EditPanel.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {memo, SyntheticEvent} from "react"
+import {CollapsibleToolbar} from "../../toolbarComponents/CollapsibleToolbar"
+import Undo from "./buttons/UndoButton"
+import Redo from "./buttons/RedoButton"
+import Layout from "./buttons/LayoutButton"
+import Copy from "./buttons/CopyButton"
+import Delete from "./buttons/DeleteButton"
+import Paste from "./buttons/PasteButton"
+import {useTranslation} from "react-i18next"
+import {useSelector} from "react-redux"
+import {getCapabilities} from "../../../reducers/selectors/other"
+import {ToolbarButtons} from "../../toolbarComponents/ToolbarButtons"
+
+type Props = {
+  selectionActions: SelectionActions,
+}
+
+export type SelectionActions = {
+  copy: (event: SyntheticEvent) => void,
+  canCopy: boolean,
+  cut: (event: SyntheticEvent) => void,
+  canCut: boolean,
+  paste: (event: SyntheticEvent) => void,
+  canPaste: boolean,
+}
+
+function EditPanel(props: Props) {
+  const capabilities = useSelector(getCapabilities)
+  const {selectionActions} = props
+  const {t} = useTranslation()
+
+  return (
+    <CollapsibleToolbar id="EDIT-PANEL" title={t("panels.edit.title", "Edit")}>
+      <ToolbarButtons small>
+        {capabilities.write ? <Undo/> : null}
```

Comment:
One `capabilities.write ` wouldn't be enough?

Reply:
yes